Clockwise/VirtualClock.cs

```diff
@@ -113,9 +122,10 @@ private VirtualClock(DateTimeOffset? now = null, [CallerMemberName] string creat
         {
             get
             {
-                for (var i = 0; i < schedule.Count; i++)
+                var length = schedule.Count;
+                for (var i = length - 1; i >= 0; i--)
```

Comment:
Scan the array backwards looking for the first timespan that is *after* the current timespan.

@jonsequitur : couldn't this just be `schedule[schedule.Count-1].Item1 - now`?

Reply:
I'm not sure what you're referring to.

Reply:
@jonsequitur Since the next action is always at the end of the list, TimeUntilNextAction could just return the last item in the list.

Reply:
The next action isn't always at the end of the list.

Reply:
In which scenarios is it not at the end?

Reply:
Actions can schedule other actions. Say you have an action A scheduled for an hour from now and another action B scheduled for a day from now. Action A, when it executes, can schedule additional actions to be executed before B becomes due.

Reply:
Oops, I got turned around. The end of the list is the beginning of the schedule now. This should be correct.